test/core/gprpp/work_serializer_test.cc

```diff
@@ -237,42 +245,47 @@ TEST(WorkSerializerTest, WorkSerializerDestructionRaceMultipleThreads) {
 
 #ifndef NDEBUG
 TEST(WorkSerializerTest, RunningInWorkSerializer) {
-  grpc_core::WorkSerializer work_serializer1;
-  grpc_core::WorkSerializer work_serializer2;
-  EXPECT_FALSE(work_serializer1.RunningInWorkSerializer());
-  EXPECT_FALSE(work_serializer2.RunningInWorkSerializer());
-  work_serializer1.Run(
-      [&]() {
-        EXPECT_TRUE(work_serializer1.RunningInWorkSerializer());
-        EXPECT_FALSE(work_serializer2.RunningInWorkSerializer());
-        work_serializer2.Run(
-            [&]() {
-              EXPECT_TRUE(work_serializer1.RunningInWorkSerializer());
-              EXPECT_TRUE(work_serializer2.RunningInWorkSerializer());
+  auto work_serializer1 =
+      std::make_shared<WorkSerializer>(GetDefaultEventEngine());
+  auto work_serializer2 =
+      std::make_shared<WorkSerializer>(GetDefaultEventEngine());
+  EXPECT_FALSE(work_serializer1->RunningInWorkSerializer());
+  EXPECT_FALSE(work_serializer2->RunningInWorkSerializer());
+  work_serializer1->Run(
```

Comment:
Doesn't this test now need a barrier of some sort to ensure that all of the WorkSerializer callbacks have actually been executed before the test ends?

Reply:
I don't think so, but added regardless